packages/atomic/src/components/insight/atomic-insight-generated-answer/atomic-insight-generated-answer.tsx

```diff
@@ -142,6 +181,62 @@ export class AtomicInsightGeneratedAnswer
     this.ariaMessage = message;
   };
 
+  private setIsCollapsed = (isCollapsed: boolean) => {
+    this.isCollapsed = isCollapsed;
+  };
+
+  private toggleClass(element: Element, className: string, condition: boolean) {
```

Comment:
This whole section of the code is suspicious: Having to manipulate classes/get reference to DOM directly like this instead of just using the render loop as normal...


Why can't we just output the correct classes on every element in the render function depending on `isCollapsed` prop ?

Reply:
For this example: `this.toggleClass(container, 'answer-collapsed', this.isCollapsed);`

Is there a reason that, as part of the `render()` loop function, is `this.isCollapsed===true`, we set the class `answer-collapsed` on the `container` ?

I don't understand why it is necessary to do it in this non-standard manner ?

Reply:
we need to manipulate the DOM regarding if collapsible option is enabled and When we resize the answer! in this case the answer is either partially hidden with show more button displayed or displayed fully if the height of the answer < 250px or when it is not collapsed with show less displayed. 

the whole logic is checked regarding these conditions and different use cases and with that we inject the right class. we have basically the classes: `answer-collapsed` and `show-button-visible` and `is-collapsible` that we add/remove.

I don't get your point, what are you suggesting to make this work ?

Reply:
Like @jelmedini, In this situation I can't say how I could handle it better than this. TBH I really dislike how all of this has been built. This common class extension pattern and DOM mutation methods makes all of this way too convoluted for my taste.

I do not want to kick a wasp nest uselessly, but while Coveo RGA capabilities are quite new, I am curious as why we choose to use a kinda already dated React class pattern that could probably have been done way cleaner with a more modern functional approach like compound Components.

That being said... without any solution to propose. I will let you folks resolve this.